lib/puppet/indirector/file_bucket_file/file.rb

```diff
@@ -28,6 +34,41 @@ def find(request)
       end
     end
 
+    def list(request)
+      fromdate = request.options[:fromdate] || "0:0:0 1-1-1970"
+      todate = request.options[:todate] || Time.now.strftime("%F %T")
+      to = Time.parse(todate)
+      from = Time.parse(fromdate)
+      bucket = Hash.new {[]}
```

Comment:
no, not with the way I populate `bucket` at line 51. It prevent additional `if` around line 51